sevntu-checks/sevntu-checks.xml

```diff
@@ -80,6 +80,10 @@
       <property name="forbiddenClassesRegexpProtected" value=""/>
       <property name="forbiddenClassesRegexpPublic" value=""/>
     </module>
+    <module name="com.github.sevntu.checkstyle.checks.coding.ForbidCertainMethods">
+      <property name="file" value="${checkstyle.forbidden_methods.file}"/>
```

Comment:
I thought we were going to try to avoid this and embed the criteria in the configuration?

Reply:
@rnveach The problem is that the [dtd](http://www.puppycrawl.com/dtds/configuration_1_2.dtd) does not allow embedding a sequence. The property can have only 3 attributes name, value & default. Other filters that had simple list have just concatenated value with space as separator. We want to be able to express regex for method name & argument count (which is optional). One way to do this can be to use a syntax such as:
```
<property name="method1.name" value="exit"/>
<property name="method2.name" value="assert(True|False)"/>
<property name="method2.argCount" value="2"/>
```
What are your thoughts ?

Reply:
@raghavgautam Yes, we can't embed DTD inside our configuration, so you have to work with the way it is current setup. If you want to exclude multiple methods, you need to define the check multiple times.

I think properties should be: `methodName` (if we are keeping it as methods) and `argumentCount` (no need to shorten it).
@romani Are you ok with this?

Reply:
Another thing to consider is that it will not fit nicely with sonar/checkstyle-extensions.xml.
https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/605/files#diff-41027f58a40e7f9572f62d5e7c281134R187

Reply:
> it will not fit nicely with sonar/checkstyle-extensions.xml.

I don't know what you mean.

Reply:
Extra config file and extra DTD file is NOT ok.
All should be in-lined to Check configuration.
If user want to catch several methods to catch, he should do several instances of Check with different configurations.
```
<module>
  <property name="method.name" value="exit"/>
  <property name="method.argCount" value="0"/>
</module>
<module>
  <property name="method.name" value="assert(True|False)"/>
  <property name="method.argCount" value="2"/>
</module>
```

Reply:
@romani @rnveach I have have modified my code to work with:
```
    <module name="com.github.sevntu.checkstyle.checks.coding.ForbidCertainMethods">
      <module name="ExitCheck">
        <property name="method.argCount" value=""/>
        <property name="method.name" value="exit"/>
        <property name="reason" value="Execution of exit can shutdown the application leading to DoS attack."/>
      </module>
    </module>
```

Reply:
I question the need for the periods, but I still think `argCount` should be fully written out to `argumentCount` like I mentioned before.

Reply:
Done.

Reply:
Done.